client/engine/engine.go

```diff
@@ -2,22 +2,24 @@
 package engine // import "github.com/statechannels/go-nitro/client/engine"
 
 import (
-	"math/big"
-
-	"github.com/statechannels/go-nitro/channel/state"
+	"github.com/statechannels/go-nitro/client/engine/chainservice"
+	"github.com/statechannels/go-nitro/client/engine/messageservice"
 	"github.com/statechannels/go-nitro/client/engine/store"
 	"github.com/statechannels/go-nitro/protocols"
-	"github.com/statechannels/go-nitro/types"
 )
 
 // Engine is the imperative part of the core business logic of a go-nitro Client
 type Engine struct {
 	// inbound go channels
-	API   chan APIEvent
-	Chain chan ChainEvent
-	Inbox chan Message
+	FromAPI   chan APIEvent // This one is exported so that the Client can send API calls
```

Comment:
todo: ToAPI chan?

Reply:
Another note: this is public to serve as a channel for a client to send events, but client code will also see the exposed named `FromAPI`, where `ToEngine` might make more sense from that perspective.

This seems like a common problem that has a conventional solution.

Reply:
About `ToAPIchan`: I think we don't want just a single `chan` for the nitro client to send to its parent application (or even for the `engine` to send to the `client`). 

The pattern instead is as follows. The application calls `client.SomeMethod()`.  The `client` then creates a dedicated `chan engine.Response` and embeds it an `APIEvent`. When the `APIEvent` is  sent it to the `engine`:
https://github.com/statechannels/go-nitro/blob/7f8e3c5cc3e7b60ff1b2891a50987094686a136b/client/client.go#L27-L36

, the engine can use it to communicate the response to `SomeMethod`. The `SomeMethod` call returns the `chan engine.Response` so that the parent application has a handle on it.

Reply:
In terms of naming the channel at each "end" (one routine sends on it, the other receives on it). I think in the above example `c.engine.FromAPI <- apiEvent` is pretty reasonable. But there are plenty of other tricks we could use, which may or may not make things more readable:

1. A simple alias
```go
	toEngine := c.engine.FromAPI
	toEngine <- apiEvent
```
2. A method on the engine that wraps the send in a function (example here uses a type switch)
```go
c.engine.SendTo(apiEvent)


func (e *Engine) SendTo(i interface{}) {
	switch i := i.(type) {
	case APIEvent:
		e.FromAPI <- i
// etc
	default:
	}
}
```
This approach would also allow us to make all of the engines channels private.

Reply:
It's also possible to get greater type safety on `chans`, by stipulating their direction https://gobyexample.com/channel-directions. AFAICT you can only do that by passing `chans` as function parameters. There's probably a way we can do that.